internal/common/function/function.go

```diff
@@ -101,6 +101,37 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup, enforcedBack
 		}
 	}
 
+	// We do this validation again just before we create the VeleroBackup object
```

Comment:
I would not run it twice

this is run prior to creating NAB, something would need to change very quickly to pass first validation and fail second

Reply:
yeah I agree with @mateusoliveira43 
If there is anything wrong with the BSL the error would be cascaded back to the NAB object status. So we are covered there.

Reply:
Ok, thanks will  update

Reply:
Simplified the logic and now the validations are ran once. Of course we are getting object to find out which BSL are we dealing with to include that in the resulting Velero Backup Object, but that happens without further validations and we are now just returning errors if k8s API gives an error.